saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/classifier/SL_model/Saul_SL_Inference.scala

```diff
@@ -95,11 +96,13 @@ class Saul_SL_Inference[HEAD <: AnyRef](factors: List[ConstrainedClassifier[_, H
             //    println("size mismatch!", cf.onClassifier.classifier.asInstanceOf[SparseNetworkLBP].getNetwork.get(i).asInstanceOf[LinearThresholdUnit].getWeightVector.size(), ",", ltuSize.size)
             //              }
             val exampleFeatureIndexes = ofDim[Int](myLTUJointlyTrainedWeight.length)
-            cf.onClassifier.classifier.asInstanceOf[SparseNetworkLBP].getLTU(i).getWeightVector.clear()
+            val weightVector = cf.onClassifier.classifier.asInstanceOf[SparseNetworkLearner].getLTU(i).asInstanceOf[LinearThresholdUnit].getWeightVector
+            weightVector.clear()
```

Comment:
I can put a breakpoint and check. I didn't verify behavior. I just forced the type of LTU to be the abstract class `LinearThresholdUnit`. This should ideally be true as the `getBaseLTU` is of the type `LinearThresholdUnit`. Let me verify this once.